### PR TITLE
[23.05] Multiple backports to OpenWrt 23.05

### PR DESCRIPTION
**Pull request by Djfe**
Some Lantiq/Kernel related small bug fixes

and two PRs that added devices

### CODE REVIEW

**Comment by Djfe** [comment]
@DragonBluep I'm thinking about backporting these as well, if you agree:
https://github.com/openwrt/openwrt/commit/4e74777fa8111a2bdf63164a4920a42917faec28
https://github.com/openwrt/openwrt/commit/285f0668f4058a935389985eb80353c8a5adbc7d

**Comment by DragonBluep** [comment]
> @DragonBluep I'm thinking about backporting these as well, if you agree: [4e74777](https://github.com/openwrt/openwrt/commit/4e74777fa8111a2bdf63164a4920a42917faec28) [285f066](https://github.com/openwrt/openwrt/commit/285f0668f4058a935389985eb80353c8a5adbc7d)

Thanks for taking care of them. I think it's worth to bockport them as they can silence some wrongly used error messages which would mislead users.

**Comment by Djfe** [comment]
I feel the same way. Thanks for creating the patches in the first place.
They are now part of this backport :)

**Comment by dfateyev** [comment]
cpufreq disable is merged to 23.05 c49654f4f0c217959542a1093e052851a3909119

**Comment by Djfe** [comment]
thx I'll update the PR

**Comment by Djfe** [comment]
Taken care of.
I added two more patches by @blocktrron

**Comment by Djfe** [comment]
Careful, I need to rebase this still (see the three failing checks)

**Comment by Djfe** [comment]
I had to refresh patches in the branch before applying my backports
I don't know whether other packages/targets might also need refreshes, I only checked the ones where the actions in this PR complained.
I could move the **first two commits** to a different PR, if you request me to do so. Also please give me feedback on commit title and message, since this is my first time refreshing patches around here.

What I ran on the bare openwrt-23.05 branch to refresh patches (so you can follow, just in-case I made a mistake):
I've selected filogic as a target and run kernel_menuconfig to build the required tools for refreshing patches
I did the same thing for x86_64 for the second commit
```
make target/linux/refresh
make package/kernel/ath10k-ct/refresh
```

Then I also had to edit @blocktrron's patch so it uses folder 5.15 instead of 6.2 of the ath10k-ct repo. Afterwards I refreshed patches again.

I'll wait till all checks complete. Then I'll add a few more backports and ask for review.

File: package/kernel/ath10k-ct/patches/988-ath10k-always-use-mac80211-loss-detection.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+sessions. The loss is communicated to the host-os, but ath10k does not
+trigger a low-ack events by itself.
+
+The mac80211 framework for loss detection however detects this
+circumstance well in case of ath10k. So use it regardless of ath10k's
+own loss detection mechanism.
+
+Signed-off-by: David Bauer <mail@david-bauer.net>
+---
+ ath10k-6.2/mac.c | 1 -
```
Comment (Djfe) [inline package/kernel/ath10k-ct/patches/988-ath10k-always-use-mac80211-loss-detection.patch:16]: just noticed this small issue. If you want me to update this to 5.15 as well, then tell me :)
It's obviously just a wrong code comment here

**Comment by Djfe** [comment]
Ok all checks went through, now to add 8 more commits.
This is ready for review now (and merge)

**Comment by blocktrron** [comment]
This is very hard to review and bring to a merge, as you touch many different parts where i suspect most maintainer are not comfortable merging it as a whole.

I advice you to split this up functionally into multiple PRs, highly increasing your chances of Reviews.

While https://github.com/openwrt/openwrt/pull/13072/commits/d0b2b1190e45b3512da11c2a2a0cd68ee9d1c0d9 has the potential of maybe interfering with the boot process, i see the chance as limited, furthermore it retains the old style in case a sep-char is not set. I see it will ease backporting of board-support later down the line, so this is an ACK.

Same for https://github.com/openwrt/openwrt/pull/13072/commits/3f25e14847cd327b7a6796dc58db1c6dbbc1ac7b

While i plan on introducing the NWA50AX Pro, I'm asking you if you've run-tested this backport with a device? I did not do so yet and will not backport such support before validating it works as intended.

**Comment by lukasz1992** [comment]
Could we backport these patches too:

https://github.com/openwrt/openwrt/commit/063641f8cf7990731919b950c1bf9eb15d1e74c9
https://github.com/openwrt/openwrt/commit/cd2b74e01e8d5f5b80b82db1cb204c13ed99dd58
https://github.com/openwrt/openwrt/commit/a110de8152df46a1e2adf7010ba75fb3b1236cd8

?

**Comment by Djfe** [comment]
Actually I was preparing a pr for nbd168 already (after writing with him in irc about it)
But separated from anything else, so he can take a look/review just his own commits once he has the time.

**Comment by hauke** [comment]
I took some of the patches from this pull request. It would be easier if you create separate pull request for separate changes.

**Review by rmilecki — CHANGES_REQUESTED** [review]
I picked some on your commits and some may be obsolete by recent changes.

Please rebase your branch so we see what's left.

**Comment by rmilecki** [comment]
@Djfe: please reopen if you find a moment to rebase your branch.

**Pull request closed without merging**